src/commonMain/kotlin/io/github/kevincianfarini/alchemist/type/Acceleration.kt

```diff
@@ -151,6 +152,17 @@ public value class Acceleration internal constructor(
      */
     public operator fun times(scale: Long): Acceleration = Acceleration(rawNanometersPerSecondSquared * scale)
 
+    /**
+     * Returns an acceleration whose value is multiplied by the specified [scale].
+     *
+     * @throws IllegalArgumentException when this acceleration is [infinite][isInfinite] and [scale] is 0.
+     */
+    public operator fun times(scale: Double): Acceleration {
+        val intScale = scale.roundToInt()
+        if (intScale.toDouble() == scale) return times(intScale)
+        return Acceleration(rawNanometersPerSecondSquared * scale)
+    }
```

Comment:
I wonder if it would make sense for div/times int/long/double to be implemented in a supertype `Quantity`, as they should to be the same across all the quantities.

Reply:
> I wonder if it would make sense for div/times int/long/double to be implemented in a supertype Quantity, as they seem to be the same across all the quantities. I can try explore that (in a separate PR)

Let's not do this. Alchemist is currently designed with the intention that some code on our side will be duplicated since that's required to reap the benefits of value classes. We may decide to not use value classes in the future, but until then we should adhere to that and not have supertypes on them (except for `Comparable`).

Reply:
Ah, they could be default implementations in an interface while remaining value classes, right?

I don't have any strong opinion here though; just came up while repeatedly copy/pasting the same implementation and changing names for each. There'll be even more duplication here with storage unit discriminators.

Reply:
For example, doing any operations on the `Quantity` supertype, would, I think require boxing of the value class.

Reply:
Well, I think this rounding code could mostly be implemented in `SaturatingLong.times(Double)`, so that would lessen the copypasta. 

> just came up while repeatedly copy/pasting the same implementation and changing names for each. There'll be even more duplication here with storage unit discriminators.

We should attempt to avoid duplication where feasible (eg. by implementing in SaturatingLong), but we shouldn't avoid duplication for the benefit of authors if it requires the values to be boxed.